pkg/collector/corechecks/containers/containerd.go

```diff
@@ -0,0 +1,359 @@
+			// For task events, we use the container ID in order to query the Tagger's API
+			tags, err := tagger.Tag(e.ID, true)
+			if err != nil {
+				// If there is an error retrieving tags from the Tagger, we can still submit the event as is.
+				log.Errorf("Could not retrieve tags for the container %s: %v", e.ID, err)
+			}
+			output.Tags = append(output.Tags, tags...)
+		}
+		sender.Event(output)
+	}
+}
+
+func computeMetrics(sender aggregator.Sender, nk context.Context, cu cutil.ContainerdItf, userTags []string) {
+	containers, err := cu.Containers()
+	if err != nil {
+		log.Errorf(err.Error())
+		return
+	}
+
+	for _, ctn := range containers {
+		t, errTask := ctn.Task(nk, nil)
+		if errTask != nil {
+			log.Tracef("Could not retrieve metrics from task %s: %s", ctn.ID()[:12], errTask.Error())
+			continue
+		}
+
+		tags, err := collectTags(ctn, nk)
+		if err != nil {
+			log.Errorf("Could not collect tags for container %s: %s", ctn.ID()[:12], err)
+		}
+		tags = append(tags, userTags...)
+		// Tagger tags
+		taggerTags, err := tagger.Tag(ctn.ID(), false)
+		if err != nil {
+			log.Errorf(err.Error())
+			continue
+		}
+		tags = append(tags, taggerTags...)
+
+		metrics, err := convertTasktoMetrics(t, nk)
+		if err != nil {
+			log.Errorf("Could not process the metrics from %s: %v", ctn.ID(), err.Error())
+			continue
+		}
+
+		err = computeExtra(sender, ctn, nk, tags)
+		if err != nil {
+			log.Errorf("Could not process metadata related metrics for %s: %v", ctn.ID()[:12], err.Error())
+		}
+
+		if metrics.Memory.Size() > 0 {
+			computeMem(sender, metrics.Memory, tags)
+		}
+
+		if metrics.CPU.Throttling != nil && metrics.CPU.Usage != nil {
+			computeCPU(sender, metrics.CPU, tags)
+		}
+
+		if metrics.Blkio.Size() > 0 {
+			computeBlkio(sender, metrics.Blkio, tags)
+		}
+
+		if len(metrics.Hugetlb) > 0 {
+			computeHugetlb(sender, metrics.Hugetlb, tags)
+		}
+	}
+}
+
+func convertTasktoMetrics(task containerd.Task, nsCtx context.Context) (*cgroups.Metrics, error) {
+	metricTask, err := task.Metrics(nsCtx)
+	if err != nil {
+		return nil, err
+	}
+
+	metricAny, err := typeurl.UnmarshalAny(&types.Any{
+		TypeUrl: metricTask.Data.TypeUrl,
+		Value:   metricTask.Data.Value,
+	})
+	if err != nil {
+		log.Errorf(err.Error())
+		return nil, err
+	}
+	return metricAny.(*cgroups.Metrics), nil
+}
+
+func computeExtra(sender aggregator.Sender, ctn containerd.Container, nsCtx context.Context, tags []string) error {
+	img, err := ctn.Image(nsCtx)
+	if err != nil {
+		return err
+	}
+	size, err := img.Size(nsCtx)
+	if err != nil {
+		return err
+	}
+	sender.Gauge("containerd.image.size", float64(size), "", tags)
+	return nil
+}
+
+func collectTags(ctn containerd.Container, nsCtx context.Context) ([]string, error) {
+	tags := []string{}
+
+	// Container image
+	im, err := ctn.Image(nsCtx)
+	if err != nil {
+		return tags, err
+	}
+	imageName := fmt.Sprintf("image:%s", im.Name())
+	tags = append(tags, imageName)
+
+	// Container labels
```

Comment:
Should we also support a label mapping configuration for consistency with k8s and docker?

Reply:
Do you mean a way to ignore containers ?
If so, I think it's reasonable and it's actually very easy to add (via the filters).
I'll add an individual commit after addressing all your feedback so you can tell me what you think.

Reply:
We'll also need to support container filtering, but I was thinking about `DOCKER_LABELS_AS_TAGS` and friends. Let's see what kube injects as labels first, they might be pretty boring.